CHANGELOG.md

```diff
@@ -1,3 +1,54 @@
+    * Calling `setState` directly in render always causes an update. This was not previously the case. Regardless, you should not be calling `setState` from render.
+    * `setState` callback (second argument) now fires immediately after `componentDidMount` / `componentDidUpdate` instead of after all components have rendered.
+  * When replacing `<A />` with `<B />`,  `B.componentWillMount` now always happens before  `A.componentWillUnmount`. Previously, `A.componentWillUnmount` could fire first in some cases.
+  * Previously, changing the `ref` to a component would always detach the ref before that component's render is called. Now, we change the `ref` later, when applying the changes to the DOM.
+  * It is not safe to re-render into a container that was modified by something other than React. This worked previously in some cases but was never supported. We now emit a warning in this case. Instead you should clean up your component trees using `ReactDOM.unmountComponentAtNode`. [See this example.](https://github.com/facebook/react/issues/10294#issuecomment-318820987)
+  * `componentDidUpdate` lifecycle no longer receives `prevContext` param. ([@bvaughn](https://github.com/bvaughn) in [#8631](https://github.com/facebook/react/pull/8631))
+  * Shallow renderer no longer calls `componentDidUpdate()` because DOM refs are not available. This also makes it consistent with `componentDidMount()` (which does not get called in previous versions either).
+  * Shallow renderer does not implement `unstable_batchedUpdates()` anymore.
+- **The names and paths to the single-file browser builds have changed to emphasize the difference between development and production builds.** For example:
+  - `react/dist/react.js` → `react/umd/react.development.js`
+  - `react/dist/react.min.js` → `react/umd/react.production.min.js`
+  - `react-dom/dist/react-dom.js` → `react-dom/umd/react-dom.development.js`
+  - `react-dom/dist/react-dom.min.js` → `react-dom/umd/react-dom.production.min.js`
+* The server renderer has been completely rewritten, with some improvements:
+  * Server rendering does not use markup validation anymore, and instead tries its best to attach to existing DOM, warning about inconsistencies. And there is no data-reactid anymore.
+  * Hydrating a server rendered container now has an explicit API. Use `ReactDOM.hydrate` instead of `ReactDOM.render` if you're reviving server rendered HTML. Keep using `ReactDOM.render` if you're just doing client-side rendering.
+* When "unknown" props are passed to DOM components, in some cases, React will now render them in the DOM. [See this post for more details.](https://facebook.github.io/react/blog/2017/09/08/dom-attributes-in-react-16.html) ([@nhunzaker](https://github.com/nhunzaker) in [#10385](https://github.com/facebook/react/pull/10385), [10564](https://github.com/facebook/react/pull/10564), [#10495](https://github.com/facebook/react/pull/10495) and others)
```

Comment:
"in some cases" sounds pretty restrictive but in practice they will almost always be passed (except for invalid values). Let's change this to "for valid values" (and keep the link to the post).

Reply:
Sounds good!